tests/unit/grains/core_test.py

```diff
@@ -28,6 +28,15 @@
 
 # Globals
 core.__salt__ = {}
+core.__opts__ = {}
```

Comment:
*Update: I know why.* Because Ubuntu is using `socket.getaddrinfo` for real and blows up on IPv6, while RH doesn't. That means that the unit test is still needs some more mocking.

@Ch3LL please mock-out `socket.getaddrinfo` for IPv4 and IPv6, so none of the OS will actually use network. We're here not to test if Python-shipped standard `getaddinfo` is working correctly, but if _our_ code is reacting right.

Hope this helps.